src/wp-includes/taxonomy.php

```diff
@@ -1531,65 +1530,75 @@ function unregister_term_meta( $taxonomy, $meta_key ) {
  *               Returns 0 if term ID 0 is passed to the function.
  */
 function term_exists( $term, $taxonomy = '', $parent = null ) {
-	global $wpdb;
-
 	if ( null === $term ) {
 		return null;
 	}
 
-	$select     = "SELECT term_id FROM $wpdb->terms as t WHERE ";
-	$tax_select = "SELECT tt.term_id, tt.term_taxonomy_id FROM $wpdb->terms AS t INNER JOIN $wpdb->term_taxonomy as tt ON tt.term_id = t.term_id WHERE ";
-
 	if ( is_int( $term ) ) {
 		if ( 0 === $term ) {
 			return 0;
 		}
-		$where = 't.term_id = %d';
+		$_term = get_term( $term, $taxonomy );
+		if ( is_wp_error( $_term ) || is_null( $_term ) ) {
+			return null;
+		}
 		if ( ! empty( $taxonomy ) ) {
-			// phpcs:ignore WordPress.DB.PreparedSQLPlaceholders.ReplacementsWrongNumber
-			return $wpdb->get_row( $wpdb->prepare( $tax_select . $where . ' AND tt.taxonomy = %s', $term, $taxonomy ), ARRAY_A );
-		} else {
-			return $wpdb->get_var( $wpdb->prepare( $select . $where, $term ) );
+			return array(
+				'term_id'          => (string) $_term->term_id,
+				'term_taxonomy_id' => (string) $_term->term_taxonomy_id,
+			);
 		}
+
+		return (string) $_term->term_id;
```

Comment:
Too bad these are all strings, but great that you did that so that it remains backward-compatible 👍